### PR TITLE
Chain agnostic role: node

### LINKED ISSUE #129: Add a chain agnostic role based on gaia
We need a new name for the main role used to set up a node.
Options:
- `cosmos`
- `node`
- `cosmos-node`

### PR DESCRIPTION
**Pull request by makeworld-the-better-one**
Fixes #129

Changing vars programmatically with commands like: `fd -H --ignore-vcs -E '.git' -t f | xargs sed -i 's/gaiad_home/chain_home/g'`

Each var change is its own commit to allow for easy diffing and spotting of mistakes.

These changes will be moved to their own role (and gaia will be untouched) before merging. For now the gaia roles is modified to allow for better diffs for review.

---
- [x] Test
- [x] Add back CI
- [x] Move into new role after reviews

### CODE REVIEW

File: docs/Playbook-Variables.md [hunk 1]
```diff
@@ -31,18 +31,18 @@ ansible-playbook gaia.yml -i examples/inventory-local.yml --extra-vars "gaiad_ve
 ## Chain Binary Installation
 | Variable               | Description                                                                     | Example Value                                                                      |
 |------------------------|---------------------------------------------------------------------------------|------------------------------------------------------------------------------------|
-| `gaiad_user`           | User account to install the chain binary in                                     | `gaia`                                                                             |
-| `gaiad_user_home`      | Path to the user account home                                                   | `/home/gaia`                                                                       |
-| `gaiad_home`           | Path to Gaia home folder home                                                   | `/home/gaia/.gaia`                                                                 |
```
Comment (dasanchez) [inline docs/Playbook-Variables.md:36]: Path to chain home folder
(the "home folder home" typo is definitely mine)

**Review by dasanchez — COMMENTED** [review]
This is excellent, great progress so far!
I added five more variables to the table:
- `gaiad_api_host` -> `chain_api_host`
- `gaiad_rpc_host` -> `chain_rpc_host`
- `gaiad_p2p_host` -> `chain_p2p_host`
- `gaiad_grpc_host` -> `chain_grpc_host`
- `gaiad_use_ssl_proxy` -> `chain_use_ssl_proxy`

There are two items that can come after the move to the `node` role, or they could even get a separate issue afterwards:
- `examples/README.md`
- `gaia_control.py` and the `gaiad_` tags it has peppered throughout the role

Good idea leaving everything in the `gaia` role for reviewing- I'm OK if you want to move things over to the `node` role now.

**Comment by makeworld-the-better-one** [comment]
> There are two items that can come after the move to the `node` role, or they could even get a separate issue afterwards:
> - `examples/README.md`
> - `gaia_control.py` and the `gaiad_` tags it has peppered throughout the role

I think the README should be a separate issue that tackles documentation of this chain agnostic setup as a whole, across all the documentation. I will update `gaia_control.py` in this PR though.

File: roles/node/tasks/apis.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+- name: Set variables for api endpoint
+  set_fact:
+    endpoint_port: '{{api_port}}'
+    site_hostname: '{{chain_api_host}}.{{inventory_hostname}}'
+  when: chain_use_ssl_proxy | default(false) | bool
+
+- name: Setup nginx SSL for gaiad api endpoint
```
Comment (dasanchez) [inline roles/node/tasks/apis.yml:8]: Remove gaiad

File: roles/node/tasks/apis.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    name: nginx-ssl
+  when: chain_use_ssl_proxy | default(false) | bool
+
+- name: Set variables for rpc endpoint
+  set_fact:
+    endpoint_port: '{{rpc_port}}'
+    site_hostname: '{{chain_rpc_host}}.{{inventory_hostname}}'
+  when: chain_use_ssl_proxy | default(false) | bool
+
+- name: Setup nginx SSL for gaiad rpc endpoint
```
Comment (dasanchez) [inline roles/node/tasks/apis.yml:19]: Remove gaiad

File: roles/node/tasks/apis.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  when: chain_use_ssl_proxy | default(false) | bool
+
+- name: Set variables for GRPC endpoint
+  set_fact:
+    endpoint_port: '{{grpc_port}}'
+    site_hostname: '{{chain_grpc_host}}.{{inventory_hostname}}'
+    grpc_vhost: true
+  when: chain_use_ssl_proxy | default(false) | bool
+
+- name: Setup nginx SSL for gaiad GRPC endpoint
```
Comment (dasanchez) [inline roles/node/tasks/apis.yml:31]: Remove gaiad

File: roles/node/tasks/apis.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    name: nginx-ssl
+  when: chain_use_ssl_proxy | default(false) | bool
+
+- name: Set variables for P2P endpoint
+  set_fact:
+    endpoint_port: '{{p2p_port}}'
+    site_hostname: '{{chain_p2p_host}}.{{inventory_hostname}}'
+  when: chain_use_ssl_proxy | default(false) | bool
+
+- name: Setup nginx SSL for gaiad P2P endpoint
```
Comment (dasanchez) [inline roles/node/tasks/apis.yml:42]: Remove gaiad

**Review by dasanchez — COMMENTED** [review]
This is great @makeworld-the-better-one 
The next step will be to replace _most_ instances of `gaiad` with a `chain_binary` or `chain_binary_name` variable, but I think this is a solid starting place.

File: node_control.py [hunk 6]
```diff
@@ -33,36 +33,36 @@
 target = args.t
 
 if operation == "restart":
-    print(os.popen("ansible-playbook gaia.yml -i " +
-                   inventory + " --tags 'gaiad_restart'").read())
+    print(os.popen("ansible-playbook node.yml -i " +
```
Comment (dasanchez) [inline node_control.py:36]: This command is missing a `target` argument.

**Review by dasanchez — CHANGES_REQUESTED** [review]
The `node_control.py` needs a `target` flag in the restart operation.

**Comment by makeworld-the-better-one** [comment]
Good catch thanks. Fixed.

**Review by dasanchez — APPROVED** [review]
Looks good!

**Comment by dasanchez** [comment]
Tested:
- Local testnet (including faucet)
  - v6.0.4
  - v7.0.0
- Theta testnet
- node_control.py: start, stop, restart, reset
- Gaia upgrade via governance:
  - v6.0.4 -> v7.0.0
  - v6.0.4 -> v7.0.3

**Pull request merged**